### DISCUSSION TITLE
Plugins (and Craft?) should be visually flagged if they have DB schema changes on the Updates page

### DISCUSSION THREAD

**Discussion by bossanova808** [craftcms/cms#8414] [Ideas]
### Description

When operating a live, user facing site - notably a Commerce site with lots of transactions coming in - and when using modern git based development practices, schema changes are tricky as we all know.

It would be a huge workflow aid if, on the Updates screen, plugins (and indeed Craft itself) - had a clear visual identification as to whether or not DB schema changes are part of the update.

(If they are not, one can generally simply update on dev and push the new code out....meaning zero down time on production.  If they do have schema changes, these updates can then be grouped for 'quiet times' when the site can be brought down).

It's an interim step along the much greater path to a real solution to the whole dev db <> production db issue, but it really would be super handy.

**Comment by brandonkelly** [comment]
Not exactly what you’re looking for, but Craft 3 has a `update/run-pending-migrations` controller action, which can be used as a post-deploy hook on your deployment service, which will run any pending DB migrations the instant the files are finished deploying.

**Comment by bossanova808** [comment]
In general, I really want to deal with addon upgrades on dev, not on the live system.  This is of course so I can test appropriately before deploying, and roll back more easily if needed.

With non-schema changing plugins, this is currently easy.  However with db changing ones, it's definitely a lot more tricky to test appropriately.  At the moment it means, in practise, I tend to fall behind on updates more than I would like, as I tend to wait until a few such updates come along (manually checking if the schema version has changed in the plugin file!) - and also until such a time as I know the site will be slow (long weekend with good weather or whatever ;) ) - and then I pop up a message, take down the site and capture the DB at that point (and it often takes a couple of hours at least to test everything properly, with so many moving parts and paths through products/checkout etc) - and do a bunch together.  If anything goes wrong I can roll back easily, restore the live site, and deal with it without too much live disruption...

A visual indication on that updates page would at least make it a lot easier to work out if any one update has a schema change or not.

Anyway - will leave it with you, & thanks for looking  - just....wanted to say that I definitely don't fancy doing updates on the live site, hook or no - inevitably I am using more plugins than I want to be (it's a big site!) - and definitely at times something goes wrong in an upgrade that breaks the site...and having that happen while live is definitely not ideal.

**Comment by narration-sd** [comment]
Just in support, would say this looks a place to put your usual fine thinking on, especially for the era of the plugin store.

If it sounds pretty handy to have even at Craft 2 level.

Sorry we are ever inventing work. I know some of it is fun, but others...and everyone has their limit of comfort to be reached once in a while.
